src/bacnet/basic/object/av.c

```diff
@@ -962,13 +966,20 @@ bool Analog_Value_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
                        object. */
                     wp_data->error_class = ERROR_CLASS_PROPERTY;
                     wp_data->error_code = ERROR_CODE_WRITE_ACCESS_DENIED;
-                } else if (Analog_Value_Present_Value_Set(
+                } else {
+                    old_value = Analog_Value_Present_Value(wp_data->object_instance);
+                    if (Analog_Value_Present_Value_Set(
                                wp_data->object_instance, value.type.Real,
                                wp_data->priority)) {
-                    status = true;
-                } else {
-                    wp_data->error_class = ERROR_CLASS_PROPERTY;
-                    wp_data->error_code = ERROR_CODE_VALUE_OUT_OF_RANGE;
+                        status = true;
+                        if (Analog_Value_Write_Present_Value_Callback) {
+                            Analog_Value_Write_Present_Value_Callback(
+                                wp_data->object_instance, old_value, value.type.Real);
```

Comment:
Although in this av.c implementation there is no priority-array, this section of code should not rely on that knowledge.  Please change value.type.Real to a call to Analog_Value_Present_Value() so that if the priority was used, it would read the correct present-value.